internal/controller/storageclaim_controller.go

```diff
@@ -435,6 +435,21 @@ func (r *StorageClaimReconciler) reconcilePhases() (reconcile.Result, error) {
 				}); err != nil {
 					return reconcile.Result{}, fmt.Errorf("failed to reconcile clientProfile: %v", err)
 				}
+			case "ClientProfileMapping":
+				clientProfileMapping := &csiopv1a1.ClientProfileMapping{}
+				clientProfileMapping.Name = r.storageClaimHash
+				clientProfileMapping.Namespace = r.OperatorNamespace
+				if _, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, clientProfileMapping, func() error {
+					if err := r.own(clientProfileMapping); err != nil {
+						return fmt.Errorf("failed to own clientProfile resource: %v", err)
+					}
+					if err := json.Unmarshal(resource.Data, &clientProfileMapping.Spec); err != nil {
```

Comment:
it would be good if we unmarshal first and if it goes successful we should own the resource WDYT?

Reply:
Followed what was there in other CreateOrUpdate

Reply:
It does not matter, the own operation only changes in memory settings. And as @rewantsoni mentioned above we usually start with the own operation.

Reply:
Ack. you can resolve this conv.